src/dropdown/dropdown.ts

```diff
@@ -93,12 +126,36 @@ export class NgbDropdown {
     }
   }
 
+  ngAfterContentInit() {
+    this._dropdownMenu = this._elementRef.nativeElement.querySelector('.dropdown-menu');
+    if (this.container === 'body') {
+      this._zoneSubscription = this._ngZone.onStable.subscribe(() => {
+        positionElements(
+            this._elementRef.nativeElement, this._dropdownMenu, this.up ? 'top-left' : 'bottom-left',
+            this.container === 'body');
+      });
+    }
+  }
+
+  ngOnDestroy() {
+    this._removeMenu();
```

Comment:
@wesleycho wouldn't it be enough to call `this.close()` and let Angular destroy nodes? This way we could drop the `_removeMenu()` method altogether, no?